src/components/HotelMoreDetails.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { fetchData } from '../utils'
+import { makeStyles } from '@material-ui/core/styles'
+import Grid from '@material-ui/core/Grid'
+import Button from '@material-ui/core/Button'
+import SingleBedIcon from '@material-ui/icons/SingleBed'
+import KingBedIcon from '@material-ui/icons/KingBed'
+import EuroIcon from '@material-ui/icons/Euro'
+import Container from '@material-ui/core/Container'
+import LoadingIcon from '../content/LoadingIcon'
+import { Link } from 'react-router-dom'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+    height: '200px',
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'center',
+    backgroundColor: 'rgb(236 236 236)',
+    margin: '5px',
+    borderRadius: '5px',
+    boxSizing: 'border-box',
+    padding: '10px',
+    '& > span': {
+      margin: theme.spacing(2),
+    },
+  },
+  icon: {
+    position: 'relative',
+    top: '6px',
+    margin: '0 5px',
+  },
+  button: {
+    position: 'relative',
+    right: '-10%',
+  },
+  loadingIcon: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+  },
+  header: {
+    fontSize: '22px',
+  },
+}))
+
+export function HotelMoreDetails(props) {
+  const classes = useStyles()
+  const [hotel, setHotel] = useState(null)
+  const [loading, setLoading] = useState(true)
+
+  const hotelId = props.match.params.id
+
+  const getHotel = async () => {
+    const hotel = await fetchData(
```

Comment:
Try catch
And before try setLoading(true)

Reply:
when I set setLoading to false and then want to change it before try I get the error